hydra-node/test/Hydra/Chain/Direct/ContractSpec.hs

```diff
@@ -223,11 +228,18 @@ healthySignature number = MultiSigned [sign sk snapshot | sk <- healthyPartyCred
  where
   snapshot = healthySnapshot{number}
 
-genCloseMutation :: (CardanoTx, Utxo) -> Gen (Text, Mutation)
+data CloseMutation
+  = MutateSignatureButNotSnapshotNumber
+  | MutateSnapshotNumberButNotSignature
+  | MutateSnapshotToIllFormedValue
+  | MutateParties
+  deriving (Generic, GenericData, Show)
+
+genCloseMutation :: (CardanoTx, Utxo) -> Gen (SomeMutationLabel, Mutation)
 genCloseMutation (_tx, _utxo) =
   oneof
-    [ second ChangeHeadRedeemer <$> genChangeHeadRedeemer
-    , second ChangeHeadDatum <$> genChangeHeadDatum
+    [ bimap SomeMutationLabel ChangeHeadRedeemer <$> genChangeHeadRedeemer
```

Comment:
Can we somehow combine the labels with the `Mutation`? It feels a bit doubled and/or maybe we can just have a simple mapping of `MutateSignatureButNotSnapshotNumber MockHead.Input -> ChangeHeadRedeemer MockHead.Input`? that way, it's a flat list of mutations and we can still leverage the enumerating labeling system you created?

Reply:
Possibly indeed, if we shove the `Mutation` inside the `SomeMutationLabel` (and rename the latter to `SomeMutation`) then it should work nicely.